server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4048,7 +4048,8 @@ public UserVmVO doInTransaction(TransactionStatus status) throws InsufficientCap
                         vm.setDetail(VmDetailConstants.FIRMWARE, "efi");
                         s_logger.info("guestOS is OSX : overwrite root disk controller to scsi, use smc and efi");
                     } else {
-                        String controllerSetting = _configDao.getValue("vmware.root.disk.controller");
+                        String controllerSetting = StringUtils.defaultIfEmpty(_configDao.getValue(Config.VmwareRootDiskControllerType.key()),
+                                Config.VmwareRootDiskControllerType.getDefaultValue());
```

Comment:
Is it the config key default value or global setting value?

Reply:
@rhtyd Didn't get you. Code change is to use the default value of the config when the value set against it is empty.

Reply:
Nevermind.